### ISSUE TITLE
New cluster fails with custom workspace-resource-id

### ISSUE THREAD

**Issue by derekperkins** [Azure/AKS#499]
My user is the owner of the log analytics workspace, yet it fails to create because of supposed permission errors.
`--workspace-resource-id /providers/Microsoft.OperationalInsights/workspaces/mylogs/`

The client 'live.com#me@me.com' with object id '{id}' has permission to perform action 'Microsoft.ContainerService/managedClusters/write' on scope '/subscriptions/{id}/resourceGroups/nozzle/providers/Microsoft.ContainerService/managedClusters/myCluster'; however, it does not have permission to perform action 'Microsoft.OperationalInsights/workspaces/sharedkeys/read' on the linked scope(s) '/providers/Microsoft.OperationalInsights/workspaces/mylogs/'.

**Comment by derekperkins** [comment]
```
$ az --version
azure-cli (2.0.38)

acr (2.0.27)
acs (2.1.1)
advisor (0.5.1)
ams (0.1.1)
appservice (0.1.35)
backup (1.1.1)
batch (3.2.4)
batchai (0.3.0)
billing (0.1.8)
cdn (0.0.14)
cloud (2.0.14)
cognitiveservices (0.1.13)
command-modules-nspkg (2.0.1)
configure (2.0.16)
consumption (0.3.1)
container (0.1.24)
core (2.0.38)
cosmosdb (0.1.21)
dla (0.1.0)
dls (0.0.22)
dms (0.0.1)
eventgrid (0.1.12)
eventhubs (0.1.3)
extension (0.0.15)
feedback (2.1.2)
find (0.2.10)
interactive (0.3.23)
iot (0.1.21)
keyvault (2.0.22)
lab (0.0.22)
maps (0.2.0)
monitor (0.1.7)
network (2.1.4)
nspkg (3.0.2)
policyinsights (0.1.0)
profile (2.0.26)
rdbms (0.2.4)
redis (0.2.13)
reservations (0.2.0)
resource (2.0.31)
role (2.0.26)
servicebus (0.1.2)
servicefabric (0.0.12)
sql (2.0.27)
storage (2.0.35)
vm (2.0.34)

Python location '/usr/local/opt/python/bin/python3.7'
Extensions directory '/Users/derek/.azure/cliextensions'

Python (Darwin) 3.7.0 (default, Jun 29 2018, 20:13:13)
[Clang 9.1.0 (clang-902.0.39.2)]
```

**Comment by derekperkins** [comment]
I tried this in Cloud Shell as well and got the same error.

**Comment by sauryadas** [comment]
@keikhara Can you please take a look?

**Comment by derekperkins** [comment]
I believe I found the cause of the error - it doesn't like the trailing slash. When I changed `--workspace-resource-id /providers/Microsoft.OperationalInsights/workspaces/mylogs/` to `--workspace-resource-id /providers/Microsoft.OperationalInsights/workspaces/mylogs`, it didn't throw that error. It seems like it should be a little more forgiving with that, but my problem appears to be solved for now. Thanks.
